server/container/util/src/main/java/org/apache/james/util/scanner/SpamAssassinInvoker.java

```diff
@@ -84,7 +87,11 @@ public SpamAssassinResult scanMail(MimeMessage message, String user) throws Mess
     }
 
     public SpamAssassinResult scanMail(MimeMessage message) throws MessagingException {
-        return scanMailWithAdditionalHeaders(message);
+        return metricFactory.withMetric(
+            "spamAssassin-check",
+            Throwing.supplier(
+                () -> scanMailWithAdditionalHeaders(message))
```

Comment:
you don't use additional headers but using this vararg method prevent you from using method reference.
could you create a method that doesn't take additional headers ?

Reply:
> could you create a method that doesn't take additional headers ?

Yes sure. It makes things more obvious.

>  using this vararg method prevent you from using method reference.

Not only. It's a Supplier, so even with that change you can not use method reference.